lib/main.js

```diff
@@ -66,6 +120,10 @@ module.exports = class Router extends EventEmitter {
         this.sink = sink ? sink : new SinkMem();
         this.metaStorage = new MetaStorage(this.sink);
 
+        if (this.options.logger) {
+            this.log = abslog(this.options.logger);
```

Comment:
just default to `console` if no `options.logger`? Then you don't have to check for `this.log` later

Reply:
Defaulting to console results in tonnes of noise, esp in tests. Kinda want that to be opt in. I guess I could pass in a noop logger in all test usages to avoid but that sounds a bit cumbersome. Good point re: checking for this.log though. Wonder if the whole check here is just unnecessary. I think if options.logger is undefined then abslog should return a noop logger.

Reply:
you can override the logger in the tests though. or mock out `console.*`